options.go

```diff
@@ -309,13 +323,23 @@ func (o *Options) Validate() error {
 	o.redirectURL, msgs = parseURL(o.RedirectURL, "redirect", msgs)
 
 	for _, u := range o.Upstreams {
+
+		// Unfortunately, url.Parse also unencodes the fragment (if any is present). Since we
+		// treat the fragment with special meaning, we don't want that. We want it interpreted
+		// literally. Let's snag it before it's modified (we'll put it back in the URL in its
+		// escaped form). If URL had something like URL.RawFragment (similar to RawPath), this
+		// wouldn't be necessary.
+		var rawFrag string
+		u, rawFrag = splitString(u, '#', true)
```

Comment:
Could you just `strings.SplitN()`? Even if you end up having to re-insert the `#` later, it would make it a lot more readable/maintainable.

Reply:
I'm not sure I see how `strings.SplitN` adds any clarity here.

Reply:
Differences are trivial. I can swap it out.